internal/k8sCommon/k8sclient/endpointslicewatcher.go

```diff
@@ -0,0 +1,307 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+package k8sclient
+
+import (
+	"fmt"
+	"sync"
+
+	"go.uber.org/zap"
+	discv1 "k8s.io/api/discovery/v1"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/tools/cache"
+)
+
+// EndpointSliceWatcher watches EndpointSlices and builds:
+//  1. ip/ip:port -> {"workload", "namespace", "node"}
+type EndpointSliceWatcher struct {
```

Comment:
nit: similar to above, seems like it can be lowercase

```suggestion
type endpointSliceWatcher struct {
```

Reply:
It needs to be since it's exported.